broadinstitute/single_cell_portal_core
# Adding support for numeric facets in Azul (SCP-3823)

This update adds support for using numeric facets like `organism_age` with XDSS requests against Azul.  This facet can be used by itself, or in conjunction with other facets.  Search requests are converting the age in years into seconds before executing, similar to how BigQuery searches are run.

In addition, this update adds significant mocking to the `AzulSearchServiceTest` and `HcaAzulClientTest` as part of an effort to reduce errors/false negatives during CI runs, usually caused by service interruptions in Azul.  Also, HCA search results now link out directly to the corresponding project in the HCA data browser.

MANUAL TESTING
Note: you may need to run `SearchFacet.update_all_facet_filters` in the Rails console if you have been working on a branch that was based off of versions of `development` before today (Nov. 22)
1. Pull this branch and sign in with an account that has `cross_dataset_search_backend` set to true
2. Run a query for `organism_age:1-5 years` and ensure the following HCA projects are returned
![Screen Shot 2021-11-22 at 3 54 19 PM](https://user-images.githubusercontent.com/729968/142934457-3ee785cb-72b2-4c5b-83e1-2a305fa1a0f6.png)
3. Refine the query to add `organ:Pancreas` and ensure only one match is returned
![Screen Shot 2021-11-22 at 3 55 14 PM](https://user-images.githubusercontent.com/729968/142934517-3483de82-4b6b-4abd-a720-3d9e5c0b7cc9.png)
4. (Optional) Click on the HCA Pancreas project title and ensure it opens on the correct page in the HCA data browser

This PR satisfies SCP-3832.
